### DISCUSSION TITLE
Explore Potential Merge with Kairos and Donation to LF Edge

### DISCUSSION THREAD

**Discussion by agracey** [rancher/elemental-toolkit#1715] [Ideas]
I was chatting with @mudler about the potential of merging Kairos and Elemental Toolkit into the same upstream project while donating to the [LF-edge](https://www.lfedge.org/). This has a positive impact for both projects as we share many underlying requirements and can likely gain wider adoption with some combined effort.

This discussion is to open up the conversation to a wider audience to collect requirements and discuss how best to move forward.

**Comment by agracey** [comment]
One thing discussed was that we should more explicitly split the "building, running, and updating the OS" from the "remote management of the OS".

**Comment by mudler** [comment]
Hi @agracey !

Kairos is an OSS, community-driven organization. As such it welcomes any contribution and partnership.  

Kairos embraces more sub-projects and components, to help in bootstrapping, managing Edge immutable deployments - Kairos is counting at the moment of ~25 active repositories. We actually were exploring the possibility to join the CNCF foundation.  

I would be more happy to help if SUSE would donate the Elemental toolkit and work together with Kairos in a joint effort, dedicating resources - we are already an community driven project, with a standalone organization and community governance. You can see Kairos maintainers listed here: https://github.com/kairos-io/kairos/blob/master/MAINTAINERS.md and our governance here: https://github.com/kairos-io/kairos/blob/master/GOVERNANCE.md.  

I'd invite you to join our office hours and connect with us, or by joining our public channels : https://github.com/kairos-io/kairos#community :wave: 

However, I'll play the devil advocate role here and be realistic - we have drifted away significantly from the Elemental toolkit to enlarge distribution support - and going back might be quite problematic. I've already talked about the technical reasons [here](https://github.com/kairos-io/kairos/discussions/429) which still are relevant as of today, and - given the lack of communication and syncronization - it's spanning even more widely with time, as I see significant changes undergoing in Elemental-cli already occurring ( for instance https://github.com/rancher/elemental-toolkit/pull/1697 and https://github.com/rancher/elemental-cli/pull/404 ).

**Reply by davidcassany** [reply]
We have been discussion internally our vision around Elemental Toolkit project and just wrote our ideas in #1716 . To expose the foundation of the of the road map we would like to take. Making the story short we essentially aim to:

* limit elemental-toolkit to its core
* decouple elemental-toolkit from package managers concepts
* keep coupled bits together

We believe this approach turns the project rather small and easy to be consumed even if different parties diverge on their Elemental based implementation. At the same time we keep Elemental Toolkit functional as a stand alone. From that point I think it would make sense having the project living under some third party organization.

**Reply by mudler** [reply]
I've read it thoroughly, and while I think this is a great plan for Elemental and how to re-design its internal layout, I think there is a strong mismatch in the Kairos long-term vision - Kairos put less importance on the package management side - as we plan to reduce the footprint of it to the bare minimum, even when building custom OS, by just using container images directly and not depending on any package-manager at distro-level.

We will steer away from the systemd/dracut dependency in the long run towards a path which is more distro-agnostic, with the ability to support natively other init systems.
